ui-tests/src/test/java/io/syndesis/qe/pages/integrations/editor/add/steps/DataMapper.java

```diff
@@ -174,14 +174,50 @@ public void selectMapping(String mappingName, SelenideElement containerElement)
         List<String> path = Arrays.asList(mappingName.trim().split("\\."));
         final String parent = path.get(0);
 
-        path.forEach(s -> {
-            SelenideElement detailElement = containerElement.find(By.id(s)).shouldBe(visible);
+        boolean previousHadNullId = false;
+
+        for( int i = 0; i < path.size(); i++ ) {
+            String s = path.get(i);
+            SelenideElement detailElement = containerElement.find(By.id(s));
+
+            if (s == null || s.equals("")) {
+                log.info("WWWWW ID is null");
+                previousHadNullId = true;
+                continue;
+            } else {
+                if (previousHadNullId) {
+                    log.info("WWWWW previous ID was null");
+                    log.info("WWWWW current  ID is  " + s);
+
+                    if (!detailElement.is(visible)) {
+                        log.info("WWWWW clicking an ancestor");
+                        containerElement.find(By.cssSelector("div[class=\"DocumentFieldDetailComponent\"")).shouldBe(visible).click();
```

Comment:
this css selector is used twice, please refactor it into the `Element` class.

Also, any particular reason to use `div[class="DocumentFieldDetailComponent"]` rather than `div.DocumentFieldDetailComponent`? Should work the same unless I'm missing something